src/Evolution/Systems/GrMhd/GhValenciaDivClean/FiniteDifference/ReconstructWork.tpp

```diff
@@ -0,0 +1,356 @@
+                                     DataVector, 3>>) {
+      // we need to handle the Wv^i reconstruction separately since we need to
+      // first compute Wv^i in the volume (it's not one of our primitives from
+      // the recovery). The components need to be stored contiguously, which is
+      // why we have the Variables `lorentz_factor_times_v_I`
+      const auto& spatial_velocity =
+          get<hydro::Tags::SpatialVelocity<DataVector, 3>>(volume_prims);
+      const auto& lorentz_factor =
+          get<hydro::Tags::LorentzFactor<DataVector>>(volume_prims);
+      lorentz_factor_times_v_I.initialize(get(lorentz_factor).size());
+      auto& volume_tensor =
+          get<hydro::Tags::LorentzFactorTimesSpatialVelocity<DataVector, 3>>(
+              lorentz_factor_times_v_I) = spatial_velocity;
+      for (size_t i = 0; i < 3; ++i) {
+        volume_tensor.get(i) *= get(lorentz_factor);
+      }
+      volume_tensor_ptr = &volume_tensor;
+    } else {
+      volume_tensor_ptr = &get<tag>(volume_prims);
+    }
+
+    const size_t number_of_variables = volume_tensor_ptr->size();
+    const gsl::span<const double> volume_vars = gsl::make_span(
+        (*volume_tensor_ptr)[0].data(), number_of_variables * volume_num_pts);
+    std::array<gsl::span<double>, 3> upper_face_vars{};
+    std::array<gsl::span<double>, 3> lower_face_vars{};
+    for (size_t i = 0; i < 3; ++i) {
+      gsl::at(upper_face_vars, i) =
+          gsl::make_span(get<tag>(gsl::at(*vars_on_upper_face, i))[0].data(),
+                         number_of_variables * reconstructed_num_pts);
+      gsl::at(lower_face_vars, i) =
+          gsl::make_span(get<tag>(gsl::at(*vars_on_lower_face, i))[0].data(),
+                         number_of_variables * reconstructed_num_pts);
+    }
+
+    DirectionMap<3, gsl::span<const double>> ghost_cell_vars{};
+    for (const auto& direction : Direction<3>::all_directions()) {
+      const auto& neighbors_in_direction = element.neighbors().at(direction);
+      ASSERT(neighbors_in_direction.size() == 1,
+             "Currently only support one neighbor in each direction, but "
+             "got "
+                 << neighbors_in_direction.size() << " in direction "
+                 << direction);
+      ghost_cell_vars[direction] = gsl::make_span(
+          &neighbor_data
+               .at(std::pair{direction, *neighbors_in_direction.begin()})
+               .data_for_reconstruction[vars_in_neighbor_count *
+                                        neighbor_num_pts],
+          number_of_variables * neighbor_num_pts);
+    }
+
+    hydro_reconstructor(make_not_null(&upper_face_vars),
+                        make_not_null(&lower_face_vars), volume_vars,
+                        ghost_cell_vars, subcell_mesh.extents(),
+                        number_of_variables);
+
+    vars_in_neighbor_count += number_of_variables;
+  });
+  tmpl::for_each<SpacetimeTagsToReconstruct>(
```

Comment:
Does the order in which I reconstruct the hydro and spacetime variables matter?

Reply:
No because the reconstruction doesn't use things like spatial metric, etc. It just needs to be done before computing the conserved variables.